### PR TITLE
mainUsers

### PR DESCRIPTION
**Pull request by avo2003**

### CODE REVIEW

File: src/users/controller/userController.ts [hunk 1]
```diff
@@ -75,14 +76,17 @@ export class UsersController {
   }
 
   @Patch('avatar')
-  @UseGuards(AuthGuard)
+   @UseGuards(AuthGuard)
```
Comment (harutyunyan611) [inline src/users/controller/userController.ts]: space

File: src/users/service/userService.ts [hunk 2]
```diff
@@ -17,23 +17,22 @@ export class UsersService {
...
-    if (!user) {
+    try {
+      const auth = await dbAuth.createUser({
+        email: createUserDto.email,
+        password: createUserDto.password
+      });
+      createUserDto.authUid = auth.uid;
+      createUserDto.avatar = process.env.AVATAR_URL;
+      const user = await this.usersRepository.create(createUserDto);
+      return user;
```
Comment (harutyunyan611) [inline src/users/service/userService.ts]: Redundant variable

**Review by harutyunyan611 — CHANGES_REQUESTED** [review]

**Review by harutyunyan611 — APPROVED** [review]

File: src/users/controller/userController.ts [hunk 3]
```diff
@@ -82,7 +83,10 @@ export class UsersController {
     @UploadedFile() file: Express.Multer.File,
     @Token() uid: string
   ): Promise<object> {
-    return this.usersService.uploadImageToCloudinary(file, uid);
```
Comment (ArthurO2002) [inline src/users/controller/userController.ts:85]: Promise cant return object

File: src/users/repository/userRepository.ts [hunk 4]
```diff
@@ -27,10 +27,10 @@ export class UserRepository {
     });
   }
 
-   findOne(id: string): Promise<User> {
-    return this.userRepository.findOne({
+  async findOne(uid: string): Promise<User> {
```
Comment (ArthurO2002) [inline src/users/repository/userRepository.ts:30]: remove async

File: src/users/repository/userRepository.ts [hunk 5]
```diff
@@ -44,7 +44,29 @@ export class UserRepository {
...
   async remove(id: number): Promise<User> {
     const removeUserId = await this.userRepository.findOne(id);
-      await dbAuth.deleteUser(removeUserId.authUid);
-      return this.userRepository.remove(removeUserId);
+    await dbAuth.deleteUser(removeUserId.authUid);
+    return this.userRepository.remove(removeUserId);
+  }
+
+  async changeSalary(id: number, salary: number): Promise<User> {
+    const changeSal = await this.userRepository.preload({
```
Comment (ArthurO2002) [inline src/users/repository/userRepository.ts:52]: shitty var name

**Pull request merged**